orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy

```diff
@@ -42,7 +42,7 @@ trait DeploymentDetailsAware {
     Closure callback) {
     Stage previousStage = getPreviousStageWithImage(stage, targetRegion, targetCloudProvider)
     def result = [:]
-    if (previousStage) {
+    if (previousStage && isCloudProviderEqual(stage, previousStage)) {
```

Comment:
Might be able to get away with just ` previousStage.context.cloudProvider == stage.context.cloudProvider` here and get rid of the function.

Reply:
i had to throw in the function to add those `null` checks , some tests pass `null` for `cloudprovider`

Reply:
what happens if:
```
Stage A          ->       Stage B                -> Stage C
findimage(aws)        findImage(titus)        deploy(aws)
```

won't stage C find image from stage B only to discard it because the cloud providers don't match?
In other words if the getPreviousStageWithImage returned the wrong image and we have to continue searching this change won't work? (I am probably missing something)

Reply:
this case works fine , i can discuss this with you offline.